src/openssl/ssl/context.cr

```diff
@@ -1,22 +1,40 @@
 class OpenSSL::SSL::Context
-  def self.default : self
-    @@default ||= new
-  end
-
   def initialize
     @handle = LibSSL.ssl_ctx_new(LibSSL.sslv23_method)
+    self.verify_mode = OpenSSL::SSL::VerifyMode::PEER
```

Comment:
Alright, the behavior in browsers is that it does prompt for a client certificate but does require it to proceed (one can cancel). I'll go with the constructor approach then.